addon/components/select-component.js

```diff
@@ -446,13 +446,28 @@ export default Ember.Component.extend(
     return this.set('selection', content.findProperty(defaultPath));
   }, 'content.[]'),
   selectableOptionsDidChange: Ember.observer(function() {
-    var highlighted;
+    const selectableOptions = this.get('selectableOptions');
+    const selection = this.get('selection');
+    const labelPath = this.get('optionLabelPath');
+    const valuePath = this.get('optionValuePath');
+    let highlighted;
+    
     if (this.get('showDropdown')) {
       highlighted = this.get('highlighted');
-      if (!this.get('selectableOptions').contains(highlighted)) {
+      if (!selectableOptions.contains(highlighted)) {
         return this.set('highlighted', this.get('selectableOptions.firstObject'));
       }
     }
+
+    // Update display name of selected option
+    if (!Ember.isNone(selection) && !Ember.isEmpty(labelPath) && !Ember.isEmpty(valuePath)) {
+      const selectedOption = 
+        selectableOptions.findBy(valuePath, Ember.get(selection, valuePath));
+
+      if (!Ember.isNone(selectedOption)) {
+        this.set(`selection.${labelPath}`, Ember.get(selectedOption, labelPath));
```

Comment:
If I'm reading this correctly, it seems like `selection` and `selectedOption` are two different objects and we're syncing them. This worries me, because they _should_ be the same object, so either the select component is doing something really wrong, or actually do have two external objects and we're mutating the state of one which we probably shouldn't be doing.

Reply:
Hmm yes, I initially thought they should be the same also, but it appears that `selection` is not the same object as the corresponding member of the `selectableOptions` array.

There is only one object passed in externally (from the array of `content` which maps to the `selectableOptions`). The `selection` is derived from the `value` parameter passed in, which is just the `optionValuePath` property of the selected option.

I'm not too familiar with what is going on internally here but I can take a closer look

Reply:
Probably good to make sure we know what's going on here. This may still be the right solution for now, until we redo Select, but it feels like it could cause a lot of bugs if there are cases where the `selection` is the same as the `value`. If it's always a clone or something then I think we're fine.

Reply:
@pzuraq Actually I think what is going on here is that I updated `selectableOptions` to a new array and the `selection` is still referencing an element in the old options array. I'll try to see if I can just instead update the labels directly in the original options array and if the changes will propagate